backend/de.metas.business/src/main/java/de/metas/product/impl/ProductDAO.java

```diff
@@ -642,4 +642,18 @@ public ImmutableList<String> retrieveSupplierApprovalNorms(@NonNull final Produc
 				.map(I_M_Product_SupplierApproval_Norm::getSupplierApproval_Norm)
 				.collect(ImmutableList.toImmutableList());
 	}
+
+	@Override
+	public ImmutableSet<ProductId> retrieveStockedProducts()
```

Comment:
rename it to `retrieveStockedProductIds`

> .createQueryBuilder(de.metas.adempiere.model.I_M_Product.class, this)

without `, this`

> .addOnlyContextClient()

remove it and better take ClientId as method param and filter by AD_Client_ID

> .addEqualsFilter(de.metas.adempiere.model.I_M_Product.COLUMNNAME_IsStocked, true)

you don't need fully qualified class 

> .orderBy().addColumn(de.metas.adempiere.model.I_M_Product.COLUMNNAME_Value)
> 				.endOrderBy()

you can write simply 
```java
.orderBy(I_M_Product.COLUMNNAME_Value)
```